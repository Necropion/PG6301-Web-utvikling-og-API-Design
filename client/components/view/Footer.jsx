const Footer = () => {
  return <div>Made by Jokubas Gedminas</div>;
};

export default Footer;
