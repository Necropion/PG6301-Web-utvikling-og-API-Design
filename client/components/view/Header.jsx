const Header = () => {
  return <h1 className="headerTitle">Simple Application</h1>;
};

export default Header;
