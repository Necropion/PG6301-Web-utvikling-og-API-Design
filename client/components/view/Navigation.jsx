import { useNavigate } from "react-router-dom";
import { useContext } from "react";
import useLoginContext from "../../hooks/useLoginContext";

const Navigation = () => {
  const navigate = useNavigate();
  const { username, googleName } = useContext(useLoginContext);

  const handleClick = async (e) => {
    e.preventDefault();

    if (e.target.value === "home") {
      navigate("/home");
    }

    if (e.target.value === "my-tasks") {
      navigate("/my-tasks");
    }

    if (e.target.value === "profile") {
      navigate("/profile");
    }

    if (e.target.value === "logout") {
      const response = await fetch("/api/login", {
        method: "DELETE",
      });

      if (!response.ok) {
        throw new Error(
          `Something went wrong ${response.status} ${response.statusText}`,
        );
      }

      if (response.ok) {
        window.sessionStorage.clear();
        navigate("/");
      }
    }
  };

  return (
    <div className="viewDiv">
      <h1 className="viewHeader">Welcome {googleName || username}</h1>
      <button className="viewBtn" value="home" onClick={handleClick}>
        Home
      </button>
      <button className="viewBtn" value="my-tasks" onClick={handleClick}>
        My Chats
      </button>
      <button className="viewBtn" value="profile" onClick={handleClick}>
        Profile
      </button>
      <button className="viewBtn" value="logout" onClick={handleClick}>
        Sign out
      </button>
    </div>
  );
};

export default Navigation;
