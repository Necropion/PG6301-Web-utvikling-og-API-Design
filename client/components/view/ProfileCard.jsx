import { useContext } from "react";
import useLoginContext from "../../hooks/useLoginContext";

const ProfileCard = () => {
  const { username, userID, googleName, googleEmail, googleImg } =
    useContext(useLoginContext);

  if (username) {
    return (
      <div className="profileDiv">
        <img
          src="https://static.vecteezy.com/system/resources/thumbnails/020/765/399/small/default-profile-account-unknown-icon-black-silhouette-free-vector.jpg"
          placeholder="No Image"
          alt="No Image"
        />
        <h3>User</h3>
        <p>{`Username: ${username}`}</p>
        <p>{`UserID: ${userID}`}</p>
      </div>
    );
  }

  if (googleName) {
    return (
      <div className="profileDiv">
        <img src={googleImg} />
        <h3>User</h3>
        <p>{`Name: ${googleName}`}</p>
        <p>{`Email: ${googleEmail}`}</p>
        <p>{"Bio: "}</p>
      </div>
    );
  }
};

export default ProfileCard;
