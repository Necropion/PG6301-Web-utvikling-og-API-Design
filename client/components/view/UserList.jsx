import { useContext } from "react";
import useLoginContext from "../../hooks/useLoginContext";
import { useNavigate } from "react-router-dom";

const UserList = () => {
  const navigate = useNavigate();
  const { userList, googleUserList } = useContext(useLoginContext);

  const handleUserClick = (username) => {
    window.sessionStorage.setItem("viewUserName", username);
    window.sessionStorage.setItem(
      "viewUserImg",
      "https://static.vecteezy.com/system/resources/thumbnails/020/765/399/small/default-profile-account-unknown-icon-black-silhouette-free-vector.jpg",
    );
    navigate("/user");
  };

  const handleGoogleUserClick = (name, img, email) => {
    window.sessionStorage.setItem("viewGoogleName", name);
    window.sessionStorage.setItem("viewGoogleImg", img);
    window.sessionStorage.setItem("viewGoogleEmail", email);
    navigate("/google-user");
  };

  return (
    <div className="searchDiv">
      <div>All Users</div>
      {userList.map((u, index) => (
        <div
          key={index}
          className="user"
          onClick={() => handleUserClick(u.username)}
        >
          <img
            src="https://static.vecteezy.com/system/resources/thumbnails/020/765/399/small/default-profile-account-unknown-icon-black-silhouette-free-vector.jpg"
            placeholder="No Image"
            alt="No Image"
          />
          <p>{u.username}</p>
        </div>
      ))}
      {googleUserList.map((g, index) => (
        <div
          key={index}
          className="user"
          onClick={() => handleGoogleUserClick(g.name, g.img, g.email)}
        >
          <img src={g.img} />
          <p>{g.name}</p>
        </div>
      ))}
    </div>
  );
};

export default UserList;
