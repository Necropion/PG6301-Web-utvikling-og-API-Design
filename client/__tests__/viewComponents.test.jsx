import renderer, { act } from "react-test-renderer";
import UserList from "../components/view/UserList";
import Footer from "../components/view/Footer";
import { MemoryRouter } from "react-router-dom";
import Header from "../components/view/Header";
import Navigation from "../components/view/Navigation";
import ProfileCard from "../components/view/ProfileCard";
import UseLoginContext from "../hooks/useLoginContext";

describe("View components tests", () => {
  it("Renders Footer", () => {
    const component = renderer.create(<Footer />);
    expect(component).toMatchSnapshot();
  });

  it("Renders Header", () => {
    const component = renderer.create(<Header />);
    expect(component).toMatchSnapshot();
  });

  it("Renders Navigation", () => {
    const component = renderer.create(
      <UseLoginContext.Provider
        value={{ username: "dummy", googleName: "dummy" }}
      >
        <MemoryRouter>
          <Navigation />
        </MemoryRouter>
      </UseLoginContext.Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  it("Renders ProfileCard", () => {
    const component = renderer.create(<ProfileCard />);
    expect(component).toMatchSnapshot();
  });

  const userList = [
    {
      username: "dummy",
    },
  ];

  const googleUserList = [
    {
      name: "dummy",
      img: "dummy",
      email: "dummy",
    },
  ];

  it("Renders UserList", () => {
    const component = renderer.create(
      <UseLoginContext.Provider value={{ userList, googleUserList }}>
        <MemoryRouter>
          <UserList />
        </MemoryRouter>
      </UseLoginContext.Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
