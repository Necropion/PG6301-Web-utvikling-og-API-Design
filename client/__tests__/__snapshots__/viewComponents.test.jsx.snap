// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View components tests Renders Footer 1`] = `
<div>
  Made by Jokubas Gedminas
</div>
`;

exports[`View components tests Renders Header 1`] = `
<h1
  className="headerTitle"
>
  Simple Application
</h1>
`;

exports[`View components tests Renders Navigation 1`] = `
<div
  className="viewDiv"
>
  <h1
    className="viewHeader"
  >
    Welcome 
    dummy
  </h1>
  <button
    className="viewBtn"
    onClick={[Function]}
    value="home"
  >
    Home
  </button>
  <button
    className="viewBtn"
    onClick={[Function]}
    value="my-tasks"
  >
    My Chats
  </button>
  <button
    className="viewBtn"
    onClick={[Function]}
    value="profile"
  >
    Profile
  </button>
  <button
    className="viewBtn"
    onClick={[Function]}
    value="logout"
  >
    Sign out
  </button>
</div>
`;

exports[`View components tests Renders ProfileCard 1`] = `null`;

exports[`View components tests Renders UserList 1`] = `
<div
  className="searchDiv"
>
  <div>
    All Users
  </div>
  <div
    className="user"
    onClick={[Function]}
  >
    <img
      alt="No Image"
      placeholder="No Image"
      src="https://static.vecteezy.com/system/resources/thumbnails/020/765/399/small/default-profile-account-unknown-icon-black-silhouette-free-vector.jpg"
    />
    <p>
      dummy
    </p>
  </div>
  <div
    className="user"
    onClick={[Function]}
  >
    <img
      src="dummy"
    />
    <p>
      dummy
    </p>
  </div>
</div>
`;
